go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -171,8 +173,40 @@ func initLocalnetGatewayInternal(nodeName string, clusterIPSubnet []string,
 
 	err = util.GatewayInit(clusterIPSubnet, nodeName, ifaceID, localnetGatewayIP,
 		macAddress, localnetGatewayNextHop, subnet, true, false, nil)
+
+	var node *kapi.Node
+	var kube kube.Interface
+	node, err = kube.GetNode(nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to get node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeGatewayMode, string(config.Gateway.Mode))
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeVlanID, strconv.Itoa(int(config.Gateway.VLANID)))
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeIfaceID, ifaceID)
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeMacAddress, macAddress)
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeSubnet, subnet)
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeLocalnetGatewayIP, localnetGatewayIP)
+	if err != nil {
+		return fmt.Errorf("failed to set annotation on node: %v", err)
+	}
+	err = kube.SetAnnotationOnNode(node, OvnNodeLocalnetGatewayNextHop, localnetGatewayNextHop)
```

Comment:
Let's have a separate function in pkg/cluster/gateway_init.go that handles setting all the annotations at one time, like `func setGatewayAnnotations(mode string, vlanid uint, ifaceID, macAddress, interfaceIP, interfaceNextHop) error` which packs the annotations into a map[string]string and then calls this new function:
```
// SetAnnotationsOnNode takes the node object and string map to set as annotations
func (k *Kube) SetAnnotationsOnNode(node *kapi.Node, annotations map[string]string) error {
	logrus.Infof("Setting annotations %v on node %s", annotations, node.Name)
	annotationData, err := json.Marshal(annotations)
	if err != nil {
		return fmt.Errorf("failed to marshal annotations: %v", err)
	}
	patchData := fmt.Sprintf(`{"metadata":{"annotations":%s}}`, annotationData)
	_, err := k.KClient.CoreV1().Nodes().Patch(node.Name, types.MergePatchType, []byte(patchData))
	if err != nil {
		logrus.Errorf("Error in setting annotation on node %s: %v", node.Name, err)
	}
	return err
}
```

that you'll add to pkg/kube/kube.go